kvcache/cache.go

```diff
@@ -29,6 +29,17 @@ type Cache interface {
 	// cache implementation used.
 	Put(ctx ml.Context, key, value ml.Tensor)
 
+	// SetConfig controls optimizations (mostly backend-specific) that may transform
+	// the output of the cache to work better with specific kernels. If not called,
+	// the backend settings will be used. This works well when calling Attention.
+	//
+	// The config can be overridden by models, especially if they require vanilla
+	// output when implementing their own version of attention. To do this, pass
+	// an empty ml.CacheConfig.
+	//
+	// Most models will not need to use this.
+	SetConfig(ml.CacheConfig)
```

Comment:
A setter is ok here, however, could we set this upon creation (maybe in a constructor)? This would also avoid the panic on double-set.

Reply:
Having it as part of the constructor is more natural but the intention is that it should be very rare that models need to set the config. In almost all cases, it will be set by the backend based on its specific requirements for fast attention.

The main reason for a model to set it is if they have a special version of attention that is not supported by our standard implementation and therefore they have written it themselves. In that case, they probably want the cache to behave in a standard way and not have based on the backend.

Given that, I didn't really want to clutter the constructor. We could have a second constructor for this case but it seemed nicer to keep the interface relatively consistent across caches.